### ISSUE TITLE
[1.10.2][SMP] Pathing causing low TPS issues mostly from Pigs and Cows

### ISSUE THREAD

**Issue by b0bst3r** [capnkirok/animania#26]
animania-1.10-0.9.8 BETA
forge:1.10.2-12.18.3.2221

TPS 12 on a brand new world with nothing built on, Nallars Tickprofiler brought up the following:

http://i.imgur.com/XN7UwRU.png
http://i.imgur.com/vQQVgUg.png - lower half 

Go to co ords and more often than not it's a cow or pig stood on the spot jumping up and down with nothing but grass in the way.  Killing the animals releases the TPS drop until it happens again.

**Comment by capnkirok** [comment]
Thank you for the precise description of the event. That really helps. Am working on pathing, and plan to have this fixed in 1.0.

**Comment by capnkirok** [comment]
Wasn't pathing... was AI causing the TPS issues. Pathing still needs to be fixed :D, but TPS should be much better in 1.0.

**Comment by b0bst3r** [comment]
Well it's better but they are still reporting high tick rates on Nallars Tick monitor, but it doesn't seem to be affecting TPS any.

**Comment by capnkirok** [comment]
Thanks, will leave this open and continue to work on it.

**Comment by ruyanve** [comment]
posting here to confirm high tick rates, mainly on pigs 
using 1.10.2 1.0.1a

**Comment by capnkirok** [comment]
What are you guys using to check TPS?

**Comment by b0bst3r** [comment]
Nallars Tick Profiler - https://jenkins.nallar.me/job/TickProfiler/job/1.10.2/

**Comment by Sunconure11** [comment]
There is also a command built into forge IIRC

**Comment by capnkirok** [comment]
After a few more updates to the AI scripts, I think this is much better. Thoughts?

![image](https://cloud.githubusercontent.com/assets/9628379/24019554/c1c17ecc-0a98-11e7-8118-85d6c9d9e127.png)

**Comment by capnkirok** [comment]
In case you are wondering what changes I made to fix this... I added a 'release valve' for all animals in case they couldn't reach their target. There is now a random chance that they will move on to the next task based on their current velocity (e.g., if they are pegged to a task, but don't seem to be moving).

Here's a shot of extreme testing. Dozens of pigs trapped in pens, unable to reach Mud, Water, or Food. 
![image](https://cloud.githubusercontent.com/assets/9628379/24032292/8e47a9b4-0ae7-11e7-9175-3b6d0b5f5453.png)
